File: 프로그래머스/lv0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.cs
using System;

public class Solution {
    public int solution(int n) {
        
        const int piece = 6;
        
        int lcm = LCM(piece, n);
        int answer = lcm / piece;
        
        return answer;
    }
    
    // 최소 공배수
    public int LCM(int a, int b)
    {
        int gcd = GCD(a, b);
        return a * b / gcd;
    }
    
    // 최대 공약수
    public int GCD(int a, int b)
    {
        if(b == 0) return a;
        return GCD(b, a % b);
    }
}

File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.cpp
#include <string>
#include <vector>

using namespace std;
int GCD(int a, int b);

int solution(int n) 
{
    return n / GCD(6, n);
}

int GCD(int a, int b)
{
    if (b == 0) return a;
    else return GCD(b, a % b);
}
